### PR TITLE
Fix no-self-use and docparams extension for async functions/methods

### PR DESCRIPTION
**Pull request by SupImDos**
## Tasks
- [x] Add yourself to CONTRIBUTORS if you are a new contributor.
- [x] Add a ChangeLog entry describing what your PR does.
- [ ] If it's a new feature, or an important bug fix, add a What's New entry in `doc/whatsnew/<current release.rst>`.
- [x] Write a good description on what the PR does.
- [x] Added tests 

## Type of Changes

|     | Type                   |
| --- | ---------------------- |
| ✓   | :bug: Bug fix          |

## Description

Fixes apparent bug with `no-self-use` and the `docparams` extension with async functions/methods.

Minimal example:
https://gist.github.com/SupImDos/6ca27b625b5cbc2aebb8ed30b2995cef

Appears that `visit_asyncfunctiondef()` and `leave_asyncfunctiondef()` were missing for those two checkers.

### CODE REVIEW

**Review by cdce8p — COMMENTED** [review]
Thank you for opening the PR! The change itself looks good already. Just a few points we still need to address before it can be merged.
- Could you add a small changelog entry?
- It seems one test is currently failing `tests/functional/a/assign/assignment_from_no_return_py3.py`. Just add `no-self-use` to the pylint disable comment.
- It would be nice if the change itself would be tested. Here would be a good place for `no-self-use`: https://github.com/PyCQA/pylint/blob/1e85bb755a4f5afcacd3b18397d5c35c820fcca1/tests/functional/n/no/no_self_use.py#L16-L18
Similar the error message needs to be added here: https://github.com/PyCQA/pylint/blob/main/tests/functional/n/no/no_self_use.txt#L1
- For `missing-param-doc` there is: https://github.com/PyCQA/pylint/blob/main/tests/functional/ext/docparams.py

**Comment by SupImDos** [comment]
@cdce8p 

Did those 4 tasks in the last 4 commits. Appears tests are passing.

Let me know if commit `31b4ce0` is not okay, revamped it a bit, but not sure if I have ruined the original intentions/spirit of those tests.

Thanks!

File: ChangeLog [hunk 1]
```diff
@@ -77,6 +77,8 @@ Release date: TBA
 
   Closes #4900
 
+# Fix ``no-self-use`` and ``docparams extension`` for async functions and methods.
```
Comment (Pierre-Sassoulas) [inline ChangeLog:80]:
```suggestion
* Fix ``no-self-use`` and ``docparams extension`` for async functions and methods.
```

**Review by Pierre-Sassoulas — APPROVED** [review]
The tests are on point, the fix is straightforward, great first contribution !

**Review by cdce8p — APPROVED** [review]
Looks good to me!
Thanks @SupImDos 🐬

**Pull request merged**